pkg/diskmaker/controllers/lvset/device_age.go

```diff
@@ -0,0 +1,31 @@
+package lvset
+
+import (
+	"sync"
+	"time"
+)
+
+var (
+	// deviceMinAge is the minimum age for a device to be considered safe to claim
+	// otherwise, it could be a device that some other entity has attached and we have not claimed.
+	deviceMinAge = time.Minute
+)
+
+func getTime() time.Time {
+	return time.Now()
+}
+
+// getFirstObserved returns the age that the key was first observed and records it if
+// it was first observed now
+func getFirstObserved(deviceAgeMap *map[string]time.Time, mux *sync.Mutex, key string, getTime func() time.Time) time.Time {
```

Comment:
I think this might be better written as a new type with mutex and map as properties of the new type. Something like:

```
type DeviceObservationAge struct{
  deviceAgeMap map[string]string
  mu *sync.Mutex
}
```

Reply:
done